### ISSUE TITLE
Setting the scaling factors of a MenuBar and a ComboBox does not apply those factors to the Placeholder widgets

### ISSUE THREAD

**Issue by CasualYT31** [texus/TGUI#200]
Regardless of whether the `MenuBar` has been assigned scaling factors, or any of its parent widgets (such as a `Group`) have been assigned scaling factors, the `MenuPlaceholder` always has a scaling factor of `(1.0f, 1.0f)`. In addition to this, once a menu has been opened, different menus can be opened by moving the mouse over the different menus, but this does not take into account the scaling factors. On the other hand, the `MenuBar` itself—and initially opening a menu—*does* take the scaling factors into account.

Images showing this behaviour. **Note:** the black dots show where my mouse pointer was at the time of the screenshot.

No Scaling Factors
![image](https://github.com/texus/TGUI/assets/21147925/158be441-2d91-4748-a011-a6bdeec32da4)

Scaling Factors on `MenuBar`
![image](https://github.com/texus/TGUI/assets/21147925/6e65846a-0eac-411b-9061-65fa62674e92)

Scaling Factors on `MenuBar` and `Group`
![image](https://github.com/texus/TGUI/assets/21147925/3875ee8b-8324-48a1-91a0-0a766fb6a061)

I can still click on the large `MenuBar` to open the menu initially, as shown here:
![image](https://github.com/texus/TGUI/assets/21147925/384fa0a2-5672-4393-892f-0932efc8ce00)

But then when I move the mouse over to where `File` is, it will not switch over to the `File` menu in the following case:
![image](https://github.com/texus/TGUI/assets/21147925/d400f660-9291-412a-b84f-ae3dd01ff628)

The menu is only switched once I move my mouse here:
![image](https://github.com/texus/TGUI/assets/21147925/0e27a6cf-b89f-4933-b2bc-4a9289dea641)

I am able to replicate this behaviour with the following code, using the latest commit on the 1.x branch (1606f54) and on the 0.9 branch (c638b49). I assume one can replicate the bug with older versions of the library, too.

```cpp
#include <TGUI/TGUI.hpp>
#include <TGUI/Backend/SFML-Graphics.hpp>

int main() {
    sf::RenderWindow window{ {800, 600}, "TGUI example - SFML_GRAPHICS backend" };
    tgui::Gui gui{ window };

    auto group = tgui::Group::create();
    auto menubar = tgui::MenuBar::create();
    menubar->addMenu("File");
    menubar->addMenuItem("Test");
    menubar->addMenu("Edit");
    menubar->addMenuItem("Test2");
    group->add(menubar);
    group->setScale(2.0f);
    menubar->setScale(2.0f);
    gui.add(group);

    while (window.isOpen()) {
        sf::Event event;
        while (window.pollEvent(event)) {
            gui.handleEvent(event);

            if (event.type == sf::Event::Closed)
                window.close();
        }
        window.clear();
        gui.draw();
        window.display();
    }
}
```

I will be spending some time trying to find a fix and/or workaround and will report any pertinent findings ASAP.

**Comment by texus** [comment]
Thanks for the detailed information.
I don't have time to further look into this today, but I was able to quickly fix it by changing MenuBar::openMenu like this:
```diff
diff --git a/src/Widgets/MenuBar.cpp b/src/Widgets/MenuBar.cpp
index a900e9b6..92ecb94c 100644
--- a/src/Widgets/MenuBar.cpp
+++ b/src/Widgets/MenuBar.cpp
@@ -602,11 +602,17 @@ namespace tgui
         if (m_parent)
         {
             // Find the RootContainer that contains the menu bar
+            Vector2f scale = getScale();
             Container* container = m_parent;
             while (container->getParent() != nullptr)
+            {
+                scale.x *= container->getScale().x;
+                scale.y *= container->getScale().y;
                 container = container->getParent();
+            }
 
             m_menuWidgetPlaceholder->setPosition(getAbsolutePosition());
+            m_menuWidgetPlaceholder->setScale(scale);
             container->add(m_menuWidgetPlaceholder, "#TGUI_INTERNAL$MenuBarMenuPlaceholder#");
         }
     }
```

However, I noticed that the position is still incorrect if the menu bar isn't located on position (0,0) in the group, so maybe there is also an issue with getAbsolutePosition() regarding the scaling.

I imagine that the same issue might exist when opening a combo box, as it creates a ListBox in a similar way to how menu bar creates its placeholder widget.

Note that if you want the entire window to be scaled then the setScale function isn't the intended method, you should change the view of the gui instead (see https://tgui.eu/tutorials/1.0/dpi-scaling/#scaling-tgui).

**Comment by CasualYT31** [comment]
Thanks for taking a look into this! I saw the same behaviour as you describe after applying the patch.

I did also create a `ComboBox` and it's true that I did see the same behaviour with that widget, too.

Thanks for redirecting me to the [Scaling TGUI](https://tgui.eu/tutorials/1.0/dpi-scaling/#scaling-tgui) page, that significantly simplifies everything :sweat_smile:! This shouldn't block me in any way now, but I should still look into making a PR for a full fix in the near future.
